packages/runtime/src/runtime.ts

```diff
@@ -174,6 +205,22 @@ export class Runtime {
     }
   }
 
+  /**
+   * mock suggested parameters according to current round
+   * @param flags payment flags
+   */
+  mockSuggestedParameters (flags: TxParams): SuggestedParams {
+    const s = mockSuggestedParams(flags);
+
+    // https://developer.algorand.org/docs/features/transactions/#setting-first-and-last-valid
+    s.firstRound = flags.firstValid ?? (this.round - 1);
+    s.lastRound = flags.firstValid === undefined || flags.validRounds === undefined
+      ? s.firstRound + 1000
+      : Number(flags.firstValid) + Number(flags.validRounds);
+
+    return s;
+  }
```

Comment:
Can't we pass `this.round` to mockSuggestedParams, and calculate `firstRound` and `lastRound` at a single place?

Reply:
yes